airflow/providers/common/sql/transfers/sql.py

```diff
@@ -0,0 +1,248 @@
+                " but its provider does not support it. Please upgrade the provider to a version that"
+                " supports `common-sql`. The hook class should be a subclass of"
+                " `airflow.providers.common.sql.hooks.sql.DbApiHook`."
+                f" Got {hook.__class__.__name__} Hook with class hierarchy: {hook.__class__.mro()}"
+            )
+
+        return hook
+
+    def get_db_hook(
+        self, conn_id: str, hook_params: Mapping | Iterable | None = None, dest: bool = False
+    ) -> DbApiHook:
+        return self._hook(conn_id, hook_params, dest)
+
+    def _transfer_data(self, src_hook, dest_hook, context: Context) -> None:
+        self.log.info("Using Common insert mode")
+        self.log.info("Querying data from source: %s", self.source_conn_id)
+        self.log.info("Executing: %s", self.source_sql)
+
+        with src_hook.get_cursor() as src_cursor:
+            if self.source_sql_parameters:
+                src_cursor.execute(self.source_sql, self.source_sql_parameters)
+            else:
+                src_cursor.execute(self.source_sql)
+
+            target_fields = [field[0] for field in src_cursor.description]
+
+            rows_total = 0
+
+            for rows in iter(lambda: src_cursor.fetchmany(self.rows_chunk), []):
+                dest_hook.insert_rows(
+                    table=self.destination_table,
+                    rows=rows,
+                    target_fields=target_fields,
+                    commit_every=self.rows_chunk,
+                )
+                rows_total += len(rows)
+
+            self.log.info("Total inserted: %s rows", rows_total)
+
+        self.log.info("Finished data transfer.")
+
+    def _oracle_tranfer_data(self, src_hook, dest_hook, context: Context) -> None:
+        self.log.info("Using Oracle bulk insert mode")
+        self.log.info("Querying data from source: %s", self.source_conn_id)
+        self.log.info("Executing: %s", self.source_sql)
+
+        with src_hook.get_cursor() as src_cursor:
+            if self.source_sql_parameters:
+                src_cursor.execute(self.source_sql, self.source_sql_parameters)
+            else:
+                src_cursor.execute(self.source_sql)
+
+            target_fields = [field[0] for field in src_cursor.description]
+
+            rows_total = 0
+
+            for rows in iter(lambda: src_cursor.fetchmany(self.rows_chunk), []):
+                dest_hook.bulk_insert_rows(
+                    table=self.destination_table,
+                    rows=rows,
+                    target_fields=target_fields,
+                    commit_every=self.rows_chunk,
+                )
+                rows_total += len(rows)
+
+            self.log.info("Total inserted: %s rows", rows_total)
+
+        self.log.info("Finished data transfer.")
+
+    def _snowflake_transfer_data(self, src_hook, dest_hook, context: Context) -> None:
```

Comment:
This one doesn't look as a generic

Reply:
@Taragolis it is better to have one common operator for various types instead of copying it into many - it contains generic execution with chunk and next you could delete oracle_to_oracle transfer (for Snowflake there is anyone)

Reply:
@Taragolis  has a point here - likely better option would be to add the methods here to hooks as `transfer_data'  methods  (if exist) and call methods here - that would be a bit more extensible - and then implementation of the `transfer_data` could then be fixed or improved just by releasing new provider version. A bit more complex change - you would need to spread the change across multiple providers, but might be worth it.

Reply:
```python
class DbApiHook:
    ...

    def transfer_source(self, *, ...) -> SomeSourceProtocol:
        msg = f"{type(self).__name__!r} not implements `transfer_source"
        raise NotImplementedError(msg)

    def transfer_destination(self, *, ...) -> None:
        msg = f"{type(self).__name__!r} not implements `transfer_destination"
        raise NotImplementedError(msg)
```

I would rather move methods into the `DbApiHook` make it abstract and implements it on specific provider hook. What current SQL implementation shows, that it doesn't works well with specific DB, and sometimes it doesn't work well in most of 20+ Hooks which implements DbApiHook

Reply:
Yes. Smth like that.